wpapi.js

```diff
@@ -411,14 +422,13 @@ WPAPI.discover = function( url ) {
 			});
 		})
 		.catch(function( err ) {
-			console.error( 'Autodiscovery failed' );
 			console.error( err );
 			if ( endpoint ) {
 				console.warn( 'Endpoint detected, proceeding despite error...' );
 				console.warn( 'Binding to ' + endpoint + ' and assuming default routes' );
 				return new WPAPI.site( endpoint );
 			}
-			return null;
+			throw new Error( 'Autodiscovery failed' );
```

Comment:
This is a breaking change, autodiscovery failure will now trigger a promise rejection rather than failing after logging an error.